app/src/components/OrderBookSnapshot.js

```diff
@@ -1,42 +1,61 @@
-import React, { Component } from 'react';
-import { withStyles, Container, Typography, FormControl, TextField, Slider, Collapse, IconButton, Card } from '@material-ui/core';
-import { Styles } from '../styles/OrderBookSnapshot';
-import { dateStringToEpoch, nanosecondsToString } from '../utils/date-utils';
-import TimestampOrderBookScroller from './TimestampOrderBookScroller';
-import classNames from 'classnames';
-import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
-
-
-import OrderBookService from '../services/OrderBookService';
-import { SNAPSHOT_INSTRUMENT, NANOSECONDS_IN_NINE_AND_A_HALF_HOURS, NANOSECONDS_IN_SIXTEEN_HOURS, CURRENT_TIME_PLACEHOLDER } from '../constants/Constants';
-
-
-class OrderBookSnapshot extends Component {
-
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            selectedDateNano: 0,
-            selectedTimeNano: 0,
-            selectedDateTimeNano: 0,
-            selectedTimeString: 'Select from slider',
-            asks: [],
-            bids: [],
-            expanded: true
-        };
-    }
-
+import React, { Component } from 'react';
+import { withStyles, Typography, FormControl, TextField, Slider, Collapse, IconButton, Tooltip, Card } from '@material-ui/core';
+import ChevronLeftSharpIcon from '@material-ui/icons/ChevronLeftSharp';
+import ChevronRightSharpIcon from '@material-ui/icons/ChevronRightSharp';
+import { Styles } from '../styles/OrderBookSnapshot';
+import { dateStringToEpoch, nanosecondsToString, epochToDateString, splitNanosecondEpochTimestamp } from '../utils/date-utils';
+import TimestampOrderBookScroller from './TimestampOrderBookScroller';
+import classNames from 'classnames';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+
+import OrderBookService from '../services/OrderBookService';
+import { SNAPSHOT_INSTRUMENT, NANOSECONDS_IN_NINE_AND_A_HALF_HOURS, NANOSECONDS_IN_SIXTEEN_HOURS } from '../constants/Constants';
+
+
+class OrderBookSnapshot extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedDateNano: 0,
+            selectedTimeNano: 0,
+            selectedDateTimeNano: 0,
+            selectedTimeString: 'Select from slider',
+            selectedDateString: '',
+            asks: [],
+            bids: [],
+            expanded: true
+        };
+    }
+
     /**
      * @desc Handles the date change for the TextField date picker
      * @param event The event object that caused the call
      */
     handleChangeDate = (event) => {
-        let selectedDateNano = parseInt(dateStringToEpoch(event.target.value + ' 00:00:00'));
+        const { selectedDateNano, selectedDateString } = this.state;
+        let value = event.target.value;
+        let newDateNano;
+        if (value === ''){ // If user clears the date input
+            this.setState({
```

Comment:
Why do you call `setState` here if the values have not changed? Won't that trigger a render?

Reply:
I wanted to disable clearing the date once it has been set. The TextField date picker now has the prop `value={selectedDateString}` so that I can set the date when we are going forwards/backwards by message and it rolls over to the next/previous day. When the user clicks on 'x' to clear the date on the date picker, it sets the value of the date picker to `''`. I could not find a way to disable the 'x' to clear the date, so my workaround was to set the state to its existing date, so that the value of the date picker would go back to what it was previously. I'm not sure if there's a better way to disable the clearing of the input.

The only solution I can think of is to allow users to clear the date.